code/modules/clothing/chameleon.dm

```diff
@@ -616,15 +653,23 @@
 	chameleon_action.emp_randomise(INFINITY)
 
 /obj/item/clothing/mask/chameleon/attack_self(mob/user)
-	vchange = !vchange
-	to_chat(user, "<span class='notice'>The voice changer is now [vchange ? "on" : "off"]!</span>")
-
+	voice_change = !voice_change
+	to_chat(user, "<span class='notice'>The voice changer is now [voice_change ? "on" : "off"]!</span>")
+
+/obj/item/clothing/mask/chameleon/get_name(mob/user)
+	var/mob/living/carbon/human/H = user
+	var/name = H.real_name
+	if(voice_change && H.wear_id)
+		var/obj/item/card/id/idcard = H.wear_id.GetID()
+		if(istype(idcard) && idcard.electric)
+			name = idcard.registered_name
+	return name
```

Comment:
Same with the review from earlier, if this is disabled then it should return null and have that handle stuff gracefully by moving on to the next checks

Reply:
I'll use a different method